protobuf/src/test/java/io/grpc/protobuf/StatusProtoTest.java

```diff
@@ -176,6 +176,15 @@ public void fromThrowable_shouldReturnNullIfNoEmbeddedStatus() {
     assertNull(StatusProto.fromThrowable(nestedSe));
   }
 
+  @Test
+  public void toStatusExceptionWithMetadataAndCause_shouldCaptureCause() {
+    StatusException se = StatusProto.toStatusException(STATUS_PROTO, new Metadata(),
+        new RuntimeException("This is a test exception."));
```

Comment:
If you save this exception, you can assert that it is the same identical exception instance.

```java
Throwable t = new RuntimeException("blah");
StatusException se = ...;

assertSame(t, se.getCause()); // assertEquals is also okay
```